bluetooth/src/androidLibMain/kotlin/scanner/Scanner.kt

```diff
@@ -62,7 +63,8 @@ actual class Scanner internal constructor(
 
     class Builder(
         private val bluetoothScanner: BluetoothLeScannerCompat = BluetoothLeScannerCompat.getScanner(),
-        private val bluetoothAdapter: BluetoothAdapter? = BluetoothAdapter.getDefaultAdapter(),
+        private val bluetoothAdapter: BluetoothAdapter? =
+            (ApplicationHolder.applicationContext.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager).adapter,
```

Comment:
I saw a deprecation here and since we have access to Application context here I went ahead and updated to what they recommend.

Reply:
:+1:

Reply:
ApplicationHolder is optional, you need to make a separate parameter for a `context`, which has a default `ApplicationHolder.applicationContext` and then use that context.

Reply:
Hi @thoutbeckers, I am not sure what you mean by optional. Reading the class and having in mind a couple of lines bellow where we set a non nullable var `private val applicationContext: Context = ApplicationHolder.applicationContext ` it seems like applicationsContext is always expected to be there as getter would throw an exception.

Reply:
yes so use that context, don't get it from `ApplicationHolder.applicationContext`, in the line you mention `ApplicationHolder.applicationContext` is only a default parameter (so optional to use).

Reply:
[121e1c6](https://github.com/splendo/kaluga/pull/501/commits/121e1c686efe91bc3d5f90cbac8922c5cc639a80) also deleted the default values from the Scanner private constructor as I saw no use (they all get overwritten in builder anyway from builder default values)